File: Matematicas/miguelito3.0.py
#https://omegaup.com/arena/problem/Miguelito-30/
import math
primerNumero, segundoNumero = input().split()
primerNumero = int(primerNumero)
segundoNumero = int(segundoNumero)
primeraRaiz = math.ceil(math.sqrt(primerNumero)) 
segundaRaiz = math.floor(math.sqrt(segundoNumero))
print(segundaRaiz- primeraRaiz + 1)

File: Matematicas/miguelito3.0.cpp
// #https://omegaup.com/arena/problem/Miguelito-30/
#include <bits/stdc++.h>

using namespace std;

int main(){
   unsigned long long int a, b;
   cin >> a >> b;
   cout << floor(sqrt(b)) - ceil(sqrt(a)) + 1;
   return 0;
}